src/primitives.es6

```diff
@@ -0,0 +1,339 @@
+  }
+
+  toValue() {
+    return this.content;
+  }
+
+  toRefract(options={}) {
+    const attributes = this.convertAttributesToRefract('toRefract');
+    const initial = {
+      element: this.element,
+      meta: this.meta,
+      attributes,
+      content: this.content
+    };
+    return _.extend(initial, options);
+  }
+
+  toCompactRefract() {
+    const attributes = this.convertAttributesToRefract('toCompactRefract');
+    return [this.element, this.meta, attributes, this.content];
+  }
+
+  /*
+   * Some attributes may be elements. This is domain-specific knowledge, so
+   * a subclass *MUST* define the attribute element names to convert. This
+   * method handles the actual serialization to refract.
+   */
+  convertAttributesToRefract(functionName) {
+    const attributes = {};
+
+    for (let name in this.attributes) {
+      if (this[attributeElementKeys].indexOf(name) !== -1) {
+        attributes[name] = this.attributes[name][functionName]();
+      } else {
+        attributes[name] = this.attributes[name];
+      }
+    }
+
+    return attributes;
+  }
+
+  /*
+   * Some attributes may be elements. This is domain-specific knowledge, so
+   * a subclass *MUST* define the attribute element names to convert. This
+   * method handles the actual conversion when loading.
+   */
+  convertAttributesToElements(conversionFunc) {
+    for (let name of this[attributeElementKeys]) {
+      if (this.attributes[name]) {
+        this.attributes[name] = conversionFunc(this.attributes[name]);
+      }
+    }
+  }
+
+  fromRefract(dom) {
+    this.element = dom.element;
+    this.meta = dom.meta;
+    this.attributes = dom.attributes;
+    this.content = dom.content;
+
+    this.convertAttributesToElements(convertFromRefract);
+
+    return this;
+  }
+
+  fromCompactRefract(tuple) {
+    this.element = tuple[0];
+    this.meta = tuple[1];
+    this.attributes = tuple[2];
+    this.content = tuple[3];
+
+    this.convertAttributesToElements(convertFromCompactRefract);
+
+    return this;
+  }
+
+  get() {
+    return this.content;
+  }
+
+  set(content) {
+    this.content = content;
+    return this;
+  }
+}
+
+export class NullType extends ElementType {
+  constructor(meta, attributes) {
+    super('null', meta, attributes, null);
+  }
+
+  set() {
+    return new Error('Cannot set the value of null');
+  }
+}
+
+export class StringType extends ElementType {
+  constructor(content, meta, attributes) {
+    super('string', content, meta, attributes);
+  }
+
+  get length() {
+    return this.content.length;
+  }
+}
+
+export class NumberType extends ElementType {
+  constructor(content, meta, attributes) {
+    super('number', content, meta, attributes);
+  }
+}
+
+export class BooleanType extends ElementType {
+  constructor(content, meta, attributes) {
+    super('boolean', content, meta, attributes);
+  }
+}
+
+class Collection extends ElementType {
+  get length() {
+    return this.content.length;
+  }
+
+  get [Symbol.iterator]() {
```

Comment:
Being relatively new to ES6, a comment here would help me know what's going on here.